9 feature create the mood inference models

XHCFS opened the pull request:
This adds both models and adds services for them.

the final implementation will be in the UI, but the text model is tested and the voice model has yet to be tested

the plan is to use the built in text to speech in android devices to create a multimodal evaluation in the case of voice

Commit:
AI: add text emotion based inference model

Commit:
configuration: add assets to pubspec.yaml

Commit:
AI: add voice based inference model

Commit:
feat: implement emotion inference services for text and voice

- Add ONNX Runtime dependency for model inference
- Implement text emotion inference using DistilBERT (GoEmotions)
- Implement voice emotion inference using DistilHuBERT (IEMOCAP)
- Add emotion mapper to map model outputs to Mood enum
- Add unified emotion inference service
- Add hierarchical emotion model system
- Add comprehensive documentation

Commit:
feat: improve tokenization with transformers package and add chunking

- Replace custom tokenization with transformers package for accurate BERT tokenization
- Add BERT vocabulary file (vocab.txt) for DistilBERT
- Implement token-based chunking for long text inputs
- Add comprehensive tokenization tests
- Add documentation for tensor names and tokenization
- Update .gitignore to exclude test files and notebooks

Commit:
fix: configure Android build to skip macOS and handle native libraries

- Configure Cargokit to only build for Android 64-bit targets (skip macOS)
- Filter Cargokit target platforms in Gradle to prevent darwin target errors
- Handle duplicate native libraries (libonnxruntime.so) in packaging
- Update CI/CD workflow to set required Cargokit environment variables
- Add build fix documentation

Commit:
chore: remove notebooks directory from repository

Commit:
feat: add real-time voice emotion detection service

- Add VoiceEmotionRecordingService for continuous emotion detection from microphone
- Implement audio recording with record package and permission handling
- Process audio in isolate to keep UI responsive during inference
- Support long recordings with result aggregation
- Optimize preprocessing (mean/variance normalization) in isolate
- Add VAD (Voice Activity Detection) with RMS and zero-crossing rate
- Fast evaluation every 4 seconds for quick feedback
- Organize services in emotion_inference folder
- Remove unnecessary documentation files
- Add Android microphone permissions

Commit:
fix: restore main.dart to default state for CI/CD builds

- Remove dependency on untracked voice_emotion_test_screen.dart
- Use default MyHomePage to ensure builds succeed

XHCFS committed:
Merge branch 'main' into 9-feature-create-the-mood-inference-models

Commit:
Fix dependency conflict: upgrade intl to ^0.20.2

- Add intl ^0.20.2 to resolve conflict with transformers ^0.0.2
- transformers >=0.0.2 requires intl ^0.20.2
- This resolves the version solving failure in CI/CD

salma-hatem commented:
There is a problem with recording permission. It doesn't request permission, so recording never works.

```
E/flutter (27358): [ERROR:flutter/runtime/dart_vm_initializer.cc(40)] Unhandled Exception: MissingPluginException(No implementation found for method hasPermission on channel com.llfbandit.record/messages)
```

XHCFS commented:
@salma-hatem I fixed the permissions issue, it should work now. And the dependency issue should be gone too. Can you rebase and merge for me :)

Commit:
fix: resolve MissingPluginException and build errors

Comprehensive fix addressing multiple issues:

1. MissingPluginException for microphone permissions:
   - Add explicit plugin registration in MainActivity with proper super call
   - Add initialization delays before using Permission.microphone and AudioRecorder
   - Improve error handling and user feedback messages
   - Add ProGuard rules to prevent code stripping issues

2. Build errors:
   - Fix error: replace MyHomePage with MainPage in main.dart
   - Fix warning: use computed colorScheme instead of hardcoded one
   - Fix duplicate permission_handler entry in pubspec.yaml
   - Add missing dependencies and assets

3. CI/CD improvements:
   - Add flutter clean step to ensure fresh builds

Commit:
fix: add assets/images/ to pubspec.yaml assets section

salma-hatem commented:
great!
